lib/models/requestInput.js

```diff
@@ -74,75 +108,19 @@ class RequestInput {
       throw new InternalError('Input request data must be a non-null object');
     }
 
-    Object.defineProperties(this, /** @memberOf RequestInput# */ {
-      jwt: {
-        enumerable: true,
-        get: function () { return _jwt.get(this); },
-        set: function (str) { _jwt.set(this, assert.assertString('jwt', str)); }
-      },
-      controller: {
-        enumerable: true,
-        get: function () { return _controller.get(this); },
-        set: function (str) {
-          // can only be set once
-          if (!_controller.get(this)) {
-            _controller.set(this, assert.assertString('controller', str));
-          }
-        }
-      },
-      action: {
-        enumerable: true,
-        get: function () { return _action.get(this); },
-        set: function (str) {
-          // can only be set once
-          if (!_action.get(this)) {
-            _action.set(this, assert.assertString('action', str));
-          }
-        }
-      },
-      resource: {
-        value: {},
-        enumerable: true
-      },
-      args: {
-        value: {},
-        enumerable: true
-      },
-      body: {
-        enumerable: true,
-        get: function () { return _body.get(this); },
-        set: function (obj) { _body.set(this, assert.assertObject('body', obj)); }
-      },
-      metadata: {
-        enumerable: true,
-        get: function () { return _metadata.get(this); },
-        set: function (obj) { _metadata.set(this, assert.assertObject('metadata', obj)); }
-      }
-    });
+    this[_jwt] = null;
+    this[_metadata] = null;
+    this[_body] = null;
+    this[_controller] = null;
+    this[_action] = null;
 
-    Object.defineProperties(this.resource, {
-      index: {
-        get: function () { return _resourceIndex.get(this); },
-        set: function(str) { _resourceIndex.set(this, assert.assertString('index', str)); },
-        enumerable: true
-      },
-      collection: {
-        get: function () { return _resourceCollection.get(this); },
-        set: function(str) { _resourceCollection.set(this, assert.assertString('collection', str)); },
-        enumerable: true
-      },
-      _id: {
-        get: function () { return _resourceId.get(this); },
-        set: function(str) { _resourceId.set(this, assert.assertString('_id', str)); },
-        enumerable: true
-      }
-    });
+    this.args = {};
+    this.resource = new Resource();
 
-    Object.seal(this.resource);
     Object.seal(this);
 
     Object.keys(data).forEach(k => {
-      if (!this.hasOwnProperty(k) && !this.resource.hasOwnProperty(k)) {
+      if (!this[`_${k}`] && !this.resource[`_${k}`]) {
```

Comment:
This is my own modification, but I have some doubts: it may not work with symbols. Please wait a bit before merging this PR, I need to make sure this works and, if not, to repair this line of code.